src/poprox_recommender/evaluation/evaluate.py

```diff
@@ -0,0 +1,148 @@
+import csv
+import gzip
+import json
+import logging
+from uuid import UUID
+
+import numpy as np
+import pandas as pd
+from docopt import docopt
+from lenskit.metrics import topn
+from tqdm import tqdm
+
+from poprox_concepts.api.recommendations import RecommendationRequest
+from poprox_concepts.domain import Article, ArticleSet
+from poprox_recommender.data.mind import MindData
+from poprox_recommender.evaluation.metrics import rank_biased_overlap
+from poprox_recommender.logging_config import setup_logging
+from poprox_recommender.paths import project_root
+
+logger = logging.getLogger("poprox_recommender.evaluate_offline")
+
+
+def convert_df_to_article_set(rec_df):
+    articles = []
+    for _, row in rec_df.iterrows():
+        articles.append(Article(article_id=row["item"], title=""))
+    return ArticleSet(articles=articles)
+
+
+def compute_rec_metric(user_recs: dict[UUID, pd.DataFrame], request: RecommendationRequest):
+    # Make sure you run generate.py first to get the recommendations data file
+    recs = user_recs[request.interest_profile.profile_id]
+    truth = mind_data.user_truth(request.interest_profile.profile_id)
+    # Convert truth index from UUID to string
+    truth.index = truth.index.astype(str)
+
+    personalized = 1 if request.interest_profile.click_history.article_ids else 0
+
+    final_rec_df = recs[recs["stage"] == "final"]
+    single_rr = topn.recip_rank(final_rec_df, truth[truth["rating"] > 0])
+    single_ndcg5 = topn.ndcg(final_rec_df, truth, k=5)
+    single_ndcg10 = topn.ndcg(final_rec_df, truth, k=10)
+
+    ranked_rec_df = recs[recs["stage"] == "ranked"]
+    ranked = convert_df_to_article_set(ranked_rec_df)
+
+    reranked_rec_df = recs[recs["stage"] == "reranked"]
+    reranked = convert_df_to_article_set(reranked_rec_df)
+
+    if ranked and reranked:
+        single_rbo5 = rank_biased_overlap(ranked, reranked, k=5)
+        single_rbo10 = rank_biased_overlap(ranked, reranked, k=10)
+    else:
+        single_rbo5 = None
+        single_rbo10 = None
+
+    return single_ndcg5, single_ndcg10, single_rr, single_rbo5, single_rbo10, personalized
+
+
+def main():
+    options = docopt(__doc__)
+    setup_logging(verbose=options["--verbose"], log_file=options["--log-file"])
+
+    global mind_data
+    mind_data = MindData()
+
+    logger.info("loading recommendations")
+    recs_fn = project_root() / "outputs" / "mind-val-recommendations.parquet"
```

Comment:
we should take both input and output files from the command line, so this can be reused.

Reply:
The command line option could just be a NAME, and if NAME is `mind-val` it expects to find `mind-val-recommendations` and produces `mind-val-user-metrics` and `mind-val-metrics`.

Reply:
We can leave this configurability to a next PR though.

Reply:
Sure, let me see if I can add to this PR (sbouldn't be too much extra work)